src/embedded.test.js

```diff
@@ -0,0 +1,608 @@
+        });
+      });
+
+      test('throws if "skipDomainVerification" is not a boolean', () => {
+        const client = new HelloSign();
+
+        expect(() => {
+          client.open(mockSignURL, {
+            clientId: mockClientId,
+            skipDomainVerification: 42,
+          });
+        }).toThrow(/"skipDomainVerification" must be a boolean/);
+      });
+
+      test('appends default value for "skip_domain_verification" to the iFrame URL if "skipDomainVerification" is not specified', (done) => {
+        const client = new HelloSign();
+
+        client.on(HelloSign.events.OPEN, (data) => {
+          const url = new URL(data.iFrameUrl);
+
+          client.close();
+
+          expect(url.searchParams.has('skip_domain_verification')).toBe(true);
+          expect(url.searchParams.get('skip_domain_verification')).toBe('0');
+          done();
+        });
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+        });
+      });
+
+      test('appends "skip_domain_verification" to the iFrame URL if "skipDomainVerification" is valid', (done) => {
+        const client = new HelloSign();
+
+        client.on(HelloSign.events.OPEN, (data) => {
+          const url = new URL(data.iFrameUrl);
+
+          client.close();
+
+          expect(url.searchParams.has('skip_domain_verification')).toBe(true);
+          expect(url.searchParams.get('skip_domain_verification')).toBe('1');
+          done();
+        });
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+          skipDomainVerification: true,
+        });
+      });
+
+      test('throws if "whiteLabeling" is not an object', () => {
+        const client = new HelloSign();
+
+        expect(() => {
+          client.open(mockSignURL, {
+            clientId: mockClientId,
+            whiteLabeling: 42,
+          });
+        }).toThrow(/"whiteLabeling" must be an object/);
+      });
+
+      test('appends "white_labeling_options" to the iFrame URL if "whiteLabeling" is valid', (done) => {
+        const client = new HelloSign();
+
+        client.on(HelloSign.events.OPEN, (data) => {
+          const url = new URL(data.iFrameUrl);
+
+          client.close();
+
+          expect(url.searchParams.has('white_labeling_options')).toBe(true);
+          expect(url.searchParams.get('white_labeling_options')).toBe(JSON.stringify({ foo: 'bar' }));
+          done();
+        });
+
+        client.open(mockSignURL, {
+          clientId: mockClientId,
+          whiteLabeling: {
+            foo: 'bar',
+          },
+        });
+      });
+    });
+
+    describe('close()', () => {
+
+      test('closes when the close button is clicked', (done) => {
+        const client = new HelloSign();
+
+        // Never visible when signing.
+        client.open(mockRequestURL, {
+          clientId: mockClientId,
+        });
+
+        const fn = jest.fn(() => {
+          expect(fn).toBeCalledTimes(1);
+          done();
+        });
```

Comment:
These four lines can be rewritten as `const fn = jest.fn(done);`.

Reply:
Fixed